src/main/java/net/minecraftforge/event/world/DimensionPreloadEvent.java

```diff
@@ -0,0 +1,33 @@
+package net.minecraftforge.event.world;
+
+import net.minecraft.world.DimensionType;
+import net.minecraftforge.common.DimensionManager;
+import net.minecraftforge.fml.common.eventhandler.Cancelable;
+import net.minecraftforge.fml.common.eventhandler.Event;
+import net.minecraftforge.common.MinecraftForge;
+
+/**
+ * This event is fired when DimensionManager is starting to init a new dimension
+ * with {@link DimensionManager#initDimension(int)},
+ * <br>
+ * This event is {@link Cancelable}.<br>
+ * <br>
+ * event is fired on the {@link MinecraftForge#EVENT_BUS}.<br>
+ * special care should be used when canceling this event
+ * use it only when you know what you are doing
```

Comment:
I will try to reformulate and hope i can explain better my resoning: this event it is meant to help with a specific problem as i tried to explain with the "mixin example above". The idea is to prevent the initDimension to turn any non 0 dimension into a WorldServerMulti.class and give the possibility to initiate another Class<? extends WorldServer> instead, so this way it would be possible to fully customize server side only creation of dimensions.
I do not know exactly how the description of this event would be more useful as i do not have a lot of experience in this matter and please if you can help me in making a better description i would be grateful